src/vmm/src/builder.rs

```diff
@@ -508,28 +554,20 @@ pub fn build_microvm_from_snapshot(
     {
         let mpidrs = construct_kvm_mpidrs(&microvm_state.vcpu_states);
         // Restore kvm vm state.
-        vmm.vm
-            .restore_state(&mpidrs, &microvm_state.vm_state)
-            .map_err(MicrovmStateError::RestoreVmState)
-            .map_err(RestoreMicrovmState)?;
+        vmm.vm.restore_state(&mpidrs, &microvm_state.vm_state)?;
     }
 
     // Restore kvm vm state.
     #[cfg(target_arch = "x86_64")]
-    vmm.vm
-        .restore_state(&microvm_state.vm_state)
-        .map_err(MicrovmStateError::RestoreVmState)
-        .map_err(RestoreMicrovmState)?;
-
-    vm_resources
-        .update_vm_config(&VmUpdateConfig {
-            vcpu_count: Some(vcpu_count),
-            mem_size_mib: Some(mem_size_mib(&guest_memory) as usize),
-            smt: Some(false),
-            cpu_template: None,
-            track_dirty_pages: Some(track_dirty_pages),
-        })
-        .map_err(SetVmResources)?;
+    vmm.vm.restore_state(&microvm_state.vm_state)?;
+
+    vm_resources.update_vm_config(&VmUpdateConfig {
+        vcpu_count: Some(vcpu_count),
+        mem_size_mib: Some(mem_size_mib(&guest_memory) as usize),
+        smt: Some(false),
+        cpu_template: None,
+        track_dirty_pages: Some(track_dirty_pages),
+    })?;
```

Comment:
General question here, why do we remove the `        .map_err(SetVmResources)?;` is it so we propagate the original error?